rachelnavarro/step/walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,112 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.Set;
 
 public final class FindMeetingQuery {
+
+  private ArrayList<TimeRange> possibleTimes = new ArrayList<TimeRange>();
+  private int currentTime = 0;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Zero out possibleTimes and currentTime with every call to query().
+    possibleTimes = new ArrayList<TimeRange>();
```

Comment:
I'm a little confused as to what these fields are supposed to do? If you zero them out every time shouldn't they just be local variables?

Reply:
@dchin2 possibleTimes and currentTime are class variables that are updated/accessed in getPossibleTimes(), scheduleMeetingBetween(), and getTimeBetween(); I made them private class variables since Java is pass by value, so this would be the easiest way to update them and save these changes. I zeroed them out in response to Andrew's previous review comment above, which recommends zeroing them out in query() because query() should look like a stateless method to the user, so if you were to call it twice with the same exact arguments you should get the same results.

Reply:
I would advise against having private class variables, especially when query() is the only public method in this class, and this class holds no state. query() should really be a static method, since it does not rely on any previous state and should be the same on every call.

I would initialize both those variables in query() and then pass them to necessary methods when they are needed. This current solution seems hacky, and in general it is best practice to restrict the use of class methods and variables to the nearest scope.

Reply:
Oh ok yes that makes sense! I just changed it @dchin2